src/main/java/de/unistuttgart/t2/common/scaling/cpu/CPUUsageManager.java

```diff
@@ -0,0 +1,79 @@
+package de.unistuttgart.t2.common.scaling.cpu;
+
+import java.util.*;
+import java.util.concurrent.*;
+
+/**
+ * Manages how much CPU is consistently used as minimum.
+ *
+ * @author Leon Hofmeister
+ * @since 1.2.0
+ */
+public final class CPUUsageManager {
```

Comment:
this seems to be a Spring project (correct me if I'm wrong), why not use a bean (@Component) here, and work with the injected instance? Then you don't need static methods, or worry about multiple instances (which makes my other comment irrelevant). Then you could do setup stuff in the normal constructor

Reply:
For the CPU approach, this is actually possible.
It was simply impossible for the memory approach as Spring doesn't seem to like `Interceptors` as injected values.
(If I remember I tried it both with `@component` and without, and the results were not good in either)
Hence, I chose this approach for consistency.
Which do you prefer?
Also, I'd say that **in this specific context**, `static` makes even more sense, because why would you want more than one instance regulating CPU and memory?
That'll only lead to incorrect results...

Reply:
I personally would prefer the spring variant, because I don't really like static methods with state.  
For example when testing, such state can be kept, even when a new spring context would be initialized.
Or if two instances want to put some load on the CPU, but with different configs (not simultaneous though).
Keeping the state spring-context specific seems cleaner to me.
Also, it would make extensibility in the future easier (projects depending on this could just provide another bean)